lib/main.rb

```diff
@@ -0,0 +1,91 @@
+#main.rb
+require_relative "planet"
+require_relative "solar_system"
+
+def user_find_distance(solar_system)
+  print "Find the distance between two planets. Planet 1: "
+  planet1_name = gets.chomp
+  print "Planet 2: "
+  planet2_name = gets.chomp
+
+  planet1 = solar_system.find_planet_by_name(planet1_name)
+  planet2 = solar_system.find_planet_by_name(planet2_name)
+
+  puts solar_system.distance_between(planet1, planet2)
+end
+
+def user_add_planet(solar_system)
+  print "Planet name: "
+  planet_name = gets.chomp
+  print "Planet color: "
+  planet_color = gets.chomp
+  print "Mass in kg: "
+  planet_mass = gets.chomp.to_i
+  print "Distance from the sun in km: "
+  planet_distance = gets.chomp.to_i
```

Comment:
If you're going to catch the user giving bogus data, you should be doing your controls for that here so that the user can try again. :)